src/transpiling/jsx_precompile.rs

```diff
@@ -367,6 +368,124 @@ fn serialize_attr(attr_name: &str, value: &str) -> String {
   format!(" {}=\"{}\"", escape_html(attr_name), escape_html(value))
 }
 
+fn merge_serializable_children(
+  children: &[JSXElementChild],
+) -> (Vec<JSXElementChild>, usize, usize) {
+  // Do a first pass over children to merge sibling text nodes
+  // and check if it contains any serializable nodes.
+  let mut text_count = 0;
+  let mut serializable_count = 0;
+  let mut buf = String::new();
+
+  let mut normalized_children: Vec<JSXElementChild> = vec![];
+  for child in children.iter() {
+    match child {
+      JSXElementChild::JSXText(jsx_text) => {
+        let text = jsx_text_to_str(jsx_text);
+        if text.is_empty() {
+          continue;
+        }
+        text_count += 1;
+
+        buf.push_str(&text);
+      }
+      JSXElementChild::JSXExprContainer(jsx_expr_container) => {
+        match &jsx_expr_container.expr {
+          // Empty JSX expressions can be ignored as they have no content
+          // Case: <div>{}</div>
+          // Case: <div>{/* fooo */}</div>
+          JSXExpr::JSXEmptyExpr(_) => {}
+          // Case: <div>{"foo"}</div>
+          // Case: <div>{2 + 2}</div>
+          JSXExpr::Expr(expr) => {
+            if let Expr::Lit(lit) = *expr.clone() {
+              match lit {
+                // Booleans are not rendered because people usually use
+                // them for conditional rendering.
+                // Case <div>{foo && <span />}</div>
+                Lit::Bool(_) => continue,
+                // Can be flattened
+                Lit::Num(num) => {
+                  let text = num.to_string();
+                  buf.push_str(&text);
+                  continue;
+                }
+                // Can be flattened
+                Lit::Str(str_lit) => {
+                  let text = str_lit.value;
```

Comment:
Then after bartek's suggestion:

```suggestion
                  let text: &str = &str_lit.value;
```